plugins/reporters/cyclonedx/src/funTest/kotlin/CycloneDxReporterFunTest.kt

```diff
@@ -99,6 +100,19 @@ class CycloneDxReporterFunTest : WordSpec({
             }
         }
 
+        "the expected XML file even if some copyrights contain non printable characters" {
```

Comment:
Would this test succeed also without the changes done in the reporter?

Or put differently: Would it make sense to have a test case which makes an assertion on the produced `Bom` instance on the actual patched up copyright statement?

Reply:
> Would this test succeed also without the changes done in the reporter?

No, because the XML file would not be written at all due to something like

```
org.cyclonedx.exception.GeneratorException: com.fasterxml.jackson.databind.JsonMappingException: Invalid white space character (0x1) in text to output
```

Reply:
The test fails without the changes in the reporter:

```
com.fasterxml.jackson.databind.JsonMappingException: Invalid white space character (0x2) in text to output (in xml 1.1, could output as a character entity) (through reference chain: org.cyclonedx.model.Bom["component"]->java.util.ArrayList[4]->org.cyclonedx.model.Component["copyright"]))
```

Therefore this test checks that the generation is possible, which is what needed to be tested.